### PR TITLE
Update doc

### PR DESCRIPTION
**Pull request by rllola**
- [x] a bit more advance structure
- [x] json rpc documentation
- [x] wasm documentation
- [x] native rust documenation (?)

### CODE REVIEW

File: docs/Main.md [hunk 1]
```diff
@@ -1,104 +1,3 @@
...
-```
-
-## Examples
-
-```js
-func myTest() {
-    return 42;
-}
-```
+Blablabla
```
Comment (jleni) [inline docs/Main.md]: Let's remove this

File: docs/jsonrpc/workflow.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+          |                        | sign_transaction()  |
+          |                        |                     |
+          |                        |-----Sentd Tx ------>|
+          |                        |                     |
+          |                        |                     |
+          |                        |<-----Notify---------|
+          |                        |                     |
+          |<------Success----------|                     |
+
+```
```
Comment (jleni) [inline docs/jsonrpc/workflow.md]: This needs to be removed and recreated as mermaid diagrams

File: docs/jsonrpc/workflow.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    Node-->>-Exchange: change event
+    Exchange->>+Node: Get_balance(address)
+    Node-->>-Exchange: balance  
+</mermaid>
+
+## Signing
+
+<mermaid>
+sequenceDiagram
+    Note over Exchange,Node: COMPLETE
```
Comment (jleni) [inline docs/jsonrpc/workflow.md]: This need to be completed

File: docs/jsonrpc/workflow.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
+# Typical workflows
+
+<!---
+Reference for mermaid
+https://mermaid-js.github.io/mermaid/#/sequenceDiagram
+-->
+
```
Comment (jleni) [inline docs/jsonrpc/workflow.md]: Maybe some additional workflows should be included?

Comment (rllola) [reply docs/jsonrpc/workflow.md]: I am thinking on how to monitor transaction... but it might changed in the future.

**Pull request merged**